xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MUCPersistenceManager.java

```diff
@@ -559,6 +562,39 @@ public static void deleteFromDB(MUCRoom room) {
         }
     }
 
+    /**
+     * Clears the chat history for a room
+     *
+     * @param room the room to cleaer chat history from
+     */
+    public static void clearRoomChatFromDB(MUCRoom room) {
+        Log.debug("Attempting to clear the chat history of room '{}' from the database.", room.getName());
+
+        if (!room.isPersistent() || !room.wasSavedToDB()) {
+            return;
+        }
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        boolean abortTransaction = false;
+        try {
+            con = DbConnectionManager.getTransactionConnection();
+            pstmt = con.prepareStatement(CLEAR_ROOM_CHAT_HISTORY);
+            pstmt.setLong(1, room.getID());
+            pstmt.executeUpdate();
+
+            // Update the room (in memory) to indicate the it's no longer in the database.
+            room.setSavedToDB(false);
```

Comment:
When the conversation log has been removed from the database, the room itself has not been deleted from the database, so I don't think that `room.setSavedToDB(false)` is correct. This would signal that the room itself no longer is in the database.

Reply:
Good spot, I've fixed this error.